File: bit_manipulation/136_single_number.cc
#include <vector>

// 2024/01/13: 自己做出来, 2min

// 利用 x ^ x = 0, x ^ 0 = x的特点.
int SingleNumber(const std::vector<int>& nums) {
  int ans = 0;
  for (int num : nums) {
    ans ^= num;
  }
  return ans;
}

File: bit_manipulation/136_single_number.py
from typing import List

# 2026/06/14: 直接写出答案, 用时 5min.
# x ^ x = 0, x ^ 0 = x
def single_number(nums: List[int]) -> int:
    res = 0
    for num in nums:
        res = res ^ num
    return res


nums = [4, 1, 2, 1, 2]
print(single_number(nums))
